src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -46,6 +46,8 @@ private struct CacheLineSeparated
         private static int s_completionCount = 0;
         private static int s_threadAdjustmentInterval;
 
+        private static volatile int s_numRequestedWorkers = 0;
+
         static ClrThreadPool()
```

Comment:
The idea is that you fetch the singleton instance once, and that do the chunk of work against that, so the extra indirection for the static access gets amortized. And accessing instance fields tends to be cheaper (depends on the exact runtime and architecture) so it maybe even win at the end.